### DISCUSSION TITLE
[Question][Hw1] Can I change signature of execute_instr?

### DISCUSSION THREAD

**Discussion by retroinspect** [prosyslab-classroom/cs424-program-reasoning#391] [Q&A]
For now, execute_instr seems to return only mutated memory.
But to implement semantics, I think it should return next label(i.e. instruction) as well.
Can I change the signature of function? Or am I missing something?

**Comment by sujin0529** [comment]
There is a way to implement it without changing the signature of `execute_instr`.
So, you can not change the signature of function.
If you want to know the next label, the function in the <b>Operations on the instructions</b> in the [llvm document](https://llvm.moe/ocaml/Llvm.html) will help you.

**Comment by KihongHeo** [comment]
Hi. I don't recommend changing the signature.
Instead, you can see [this document](https://github.com/prosyslab-classroom/llvm-primer/blob/master/data-structure.md#instructions-within-a-block).

To help students who are not familiar with LLVM, we provide the [LLVM Primer](https://github.com/prosyslab-classroom/llvm-primer/). Please read the document.
